AlfrescoDocumentPickerFileProvider/AFPFileService.m

```diff
@@ -0,0 +1,150 @@
+                                            [[AFPDataManager sharedManager] removeItemMetadataForIdentifier:item.identifier];
+                                        }
+                                    }
+                                    else
+                                    {
+                                        NSLog(@"==== this is error %@", error);
+                                    }
+                                    networkOperationCallbackComplete = YES;
+                                } progressBlock:nil];
+                            }
+                            else if (error.code == kAlfrescoErrorCodeRequestedNodeNotFound)
+                            {
+                                // copy to local files as parent node was not found on server
+                                [self saveToLocalFilesDocumentAtURL:[NSURL fileURLWithPath:itemMetadata.filePath]];
+                                networkOperationCallbackComplete = YES;
+                            }
+                        }];
+                    }
+                    else
+                    {
+                        networkOperationCallbackComplete = YES;
+                    }
+                }
+                else
+                {
+                    networkOperationCallbackComplete = YES;
+                }
+            }];
+            
+            /*
+             * Keep this object around long enough for the network operations to complete.
+             * Running as a background thread, seperate from the UI, so should not cause
+             * Any issues when blocking the thread.
+             */
+            do
+            {
+                [[NSRunLoop currentRunLoop] runMode:NSDefaultRunLoopMode beforeDate:[NSDate distantFuture]];
+            }
+            while (networkOperationCallbackComplete == NO);
+        }
+
+    });
+}
+
+- (NSURL *)localFilesURLForFilename:(NSString *)filename
+{
+    NSString *downloadContentPath = [[AlfrescoFileManager sharedManager] downloadsContentFolderPath];
+    NSString *fullDestinationPath = [downloadContentPath stringByAppendingPathComponent:filename];
+    NSURL *destinationURL = [NSURL fileURLWithPath:fullDestinationPath];
+    return destinationURL;
+}
+
+- (void)saveToLocalFilesDocumentAtURL:(NSURL *)url
+{
+    NSError *error = nil;
+    NSDictionary *resourceValues = [url resourceValuesForKeys:@[NSURLNameKey, NSURLCreationDateKey, NSURLContentModificationDateKey, NSURLTypeIdentifierKey, NSURLTotalFileSizeKey] error:&error];
+    NSURL * destinationURL = [self localFilesURLForFilename:resourceValues[NSURLNameKey]];
+    [self saveDocumentAtURL:url toURL:destinationURL overwritingExistingFile:NO];
+}
+
+- (NSError *)saveDocumentAtURL:(NSURL *)readingURL toURL:(NSURL *)writingURL overwritingExistingFile:(BOOL)shouldOverwrite
+{
+    NSError *copyError = nil;
+    NSFileManager *fileManager = [[NSFileManager alloc] init];
```

Comment:
It's not wrong but you could use the defaultManager convenience method to avoid allocating a new vanilla NSFileManager